Postgres 数据库 Update 报错

Golang: `1.8.2` 
xorm: 最新版

---

我定义一个表：`tb_user_info`
只有cmd工具生成模型如下
```	
type TbUserInfo struct {
	Id              int64     `xorm:"pk autoincr unique BIGINT" json:"id"`
	Phone           string    `xorm:"not null unique VARCHAR(20)" json:"phone"`
	UserName        string    `xorm:"VARCHAR(20)" json:"user_name"`
	Gender          int       `xorm:"default 0 INTEGER" json:"gender"`
	Pw              string    `xorm:"VARCHAR(100)" json:"pw"`
	Token           string    `xorm:"TEXT" json:"token"`
	Avatar          string    `xorm:"TEXT" json:"avatar"`
	Extras          interface{}    `xorm:"JSON" json:"extras"`
	Created         time.Time `xorm:"DATETIME created"`
	Updated         time.Time `xorm:"DATETIME updated"`
	Deleted         time.Time `xorm:"DATETIME deleted"`
}
```
    
然后我想更新用户手机号为 `13126564922` 的用户的 `token` 为 `ABCDEFG`
    
方法一: 
```
targetUsr := model.TbUserInfo{Phone:"13126564922"}
changeUsr := model.TbUserInfo{Token:"ABCDEFG"}
af,err := model.GDAO.Eng.Update(&changeUsr, &targetUsr)
```

方法二:
```
changeUsr := model.TbUserInfo{Token:"ABCDEFG"}
af,err := model.GDAO.Eng.Table(new(model.TbUserInfo)).Where("phone=?", 13126564922).Update(&changeUsr)
```

方法三：
```
changeUsr := model.TbUserInfo{Token:"ABCDEFG"}
af,err := model.GDAO.Eng.Table(new(model.TbUserInfo)).ID(1).Update(&changeUsr)
```

---

都是报错，无法更新：
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa0 pc=0x12b78fc]

goroutine 1 [running]:
github.com/go-xorm/xorm.buildUpdates(0xc4200cf2c0, 0xc42000a800, 0x1382480, 0xc4200da2a0, 0x0, 0xc420017b90, 0xc420017bc0, 0xc420017b60, 0x1710001, 0x0, ...)
	/Users/aboo/Golang/GOPATH/src/github.com/go-xorm/xorm/statement.go:321 +0xa7c
github.com/go-xorm/xorm.(*Session).Update(0xc420112000, 0x1382480, 0xc4200da2a0, 0xc420049bd8, 0x1, 0x1, 0x0, 0x0, 0x0)
	/Users/aboo/Golang/GOPATH/src/github.com/go-xorm/xorm/session_update.go:184 +0x4223
github.com/go-xorm/xorm.(*Engine).Update(0xc4200cf2c0, 0x1382480, 0xc4200da2a0, 0xc420049bd8, 0x1, 0x1, 0x0, 0x0, 0x0)
	/Users/aboo/Golang/GOPATH/src/github.com/go-xorm/xorm/engine.go:1411 +0xc4
main.updateUser()
	/Users/aboo/GoglandProjects/XormDemo/main.go:141 +0x186
main.main()
	/Users/aboo/GoglandProjects/XormDemo/main.go:32 +0x20
exit status 2
```


---

唯有使用 map 传入更新字段才可以完成
```
info := map[string]string{"token":"ABCDEFG"}
af,err := model.GDAO.Eng.Table(new(model.TbUserInfo)).Where("phone=?", 13126564922).Update(info)
```


---

想问一下这是怎么回事？
PG 数据库自能用 map 的方式 Update 吗？
谢谢，希望能解答一下😊

--- a/session_update_test.go
+++ b/session_update_test.go
@@ -1123,3 +1123,33 @@ func TestNoUpdate(t *testing.T) {
 	assert.Error(t, err)
 	assert.EqualValues(t, "No content found to be updated", err.Error())
 }
+
+func TestNewUpdate(t *testing.T) {
+	assert.NoError(t, prepareEngine())
+
+	type TbUserInfo struct {
+		Id       int64       `xorm:"pk autoincr unique BIGINT" json:"id"`
+		Phone    string      `xorm:"not null unique VARCHAR(20)" json:"phone"`
+		UserName string      `xorm:"VARCHAR(20)" json:"user_name"`
+		Gender   int         `xorm:"default 0 INTEGER" json:"gender"`
+		Pw       string      `xorm:"VARCHAR(100)" json:"pw"`
+		Token    string      `xorm:"TEXT" json:"token"`
+		Avatar   string      `xorm:"TEXT" json:"avatar"`
+		Extras   interface{} `xorm:"JSON" json:"extras"`
+		Created  time.Time   `xorm:"DATETIME created"`
+		Updated  time.Time   `xorm:"DATETIME updated"`
+		Deleted  time.Time   `xorm:"DATETIME deleted"`
+	}
+
+	assertSync(t, new(TbUserInfo))
+
+	targetUsr := TbUserInfo{Phone: "13126564922"}
+	changeUsr := TbUserInfo{Token: "ABCDEFG"}
+	af, err := testEngine.Update(&changeUsr, &targetUsr)
+	assert.NoError(t, err)
+	assert.EqualValues(t, 0, af)
+
+	af, err = testEngine.Table(new(TbUserInfo)).Where("phone=?", 13126564922).Update(&changeUsr)
+	assert.NoError(t, err)
+	assert.EqualValues(t, 0, af)
+}

--- a/statement.go
+++ b/statement.go
@@ -272,6 +272,9 @@ func buildUpdates(engine *Engine, table *core.Table, bean interface{},
 
 		fieldValue := *fieldValuePtr
 		fieldType := reflect.TypeOf(fieldValue.Interface())
+		if fieldType == nil {
+			continue
+		}
 
 		requiredField := useAllCols
 		includeNil := useAllCols

